Code/GraphMol/hanoitest.cpp

```diff
@@ -76,22 +79,19 @@ void qs1(const std::vector<std::vector<int>> &vects) {
 void hs1(const std::vector<std::vector<int>> &vects) {
   BOOST_LOG(rdInfoLog) << "sorting (hanoi sort) vectors" << std::endl;
   for (const auto &vect : vects) {
-    const int *data = &vect.front();
+    const int *data = vect.data();
     int_compare_ftor icmp(data);
-    int *indices = (int *)malloc(vect.size() * sizeof(int));
+    std::vector<int> indices(vect.size());
+    std::span<int> ispan(indices);
     for (unsigned int j = 0; j < vect.size(); ++j) {
       indices[j] = j;
     }
-    int *count = (int *)malloc(vect.size() * sizeof(int));
-    int *changed = (int *)malloc(vect.size() * sizeof(int));
-    memset(changed, 1, vect.size() * sizeof(int));
-    RDKit::hanoisort(indices, vect.size(), count, changed, icmp);
+    std::vector<int> count(vect.size());
+    std::vector<int> changed(vect.size(), 1);
+    RDKit::hanoisort(ispan, vect.size(), count, changed, icmp);
```

Comment:
This is "correct" but I think it should be ispan.size() not vect.size(), or better yet, just take the span and have hanoisort deduce the size?